webpack/scenes/ContentViews/Publish/CVPublishFinish.js

```diff
@@ -0,0 +1,147 @@
+import getContentViewDetails, { pollPublishTask, handleFinishedTask } from '../Details/ContentViewDetailActions';
+import Loading from '../../../components/Loading';
+import EmptyStateMessage from '../../../components/Table/EmptyStateMessage';
+import { cvVersionPublishKey } from '../ContentViewsConstants';
+
+
+const CVPublishFinish = ({
+  cvId,
+  checkedItems, setCheckedItems,
+  forcePromote, description, setDescription,
+  setIsOpen, versionCount,
+}) => {
+  const dispatch = useDispatch();
+  const [redirect, setRedirect] = useState(false);
+  const [saving, setSaving] = useState(true);
+  const [polling, setPolling] = useState(false);
+  const [taskErrored, setTaskErrored] = useState(false);
+  const response = useSelector(state => selectPublishContentViews(state, cvId, versionCount));
+  const status = useSelector(state => selectPublishContentViewStatus(state, cvId, versionCount));
+  const error = useSelector(state => selectPublishContentViewsError(state, cvId, versionCount));
+  const pollResponse = useSelector(state =>
+    selectPublishTaskPoll(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+  const pollResponseStatus = useSelector(state =>
+    selectPublishTaskPollStatus(state, cvVersionPublishKey(cvId, versionCount)), shallowEqual);
+
+  useEffect(() => {
+    setSaving(true);
+    dispatch(publishContentView({
+      id: cvId,
+      versionCount,
+      description,
+      environment_ids: checkedItems.map(item => item.id),
+      is_force_promote: (forcePromote.length > 0),
+    }));
+    setDescription('');
+    setCheckedItems([]);
+  }, [dispatch, setSaving, setDescription, setCheckedItems]);
+
+  useEffect(() => {
+    setSaving(true);
+    const { id } = response;
+    if (id && status === STATUS.RESOLVED) {
+      setSaving(false);
+      dispatch(pollPublishTask(cvVersionPublishKey(cvId, versionCount), response));
+      setPolling(true);
+    } else if (status === STATUS.ERROR) {
+      setSaving(false);
+    }
+  }, [JSON.stringify(response), status, error]);
+
+  useEffect(() => {
+    const { state, result } = pollResponse;
+    if (response && (state === 'paused' || result === 'error')) {
+      setTaskErrored(true);
+    }
+    if (response && (state === 'stopped' && result === 'success')) {
+      setPolling(false);
+      dispatch(handleFinishedTask(cvId, cvVersionPublishKey(cvId, versionCount), pollResponse));
+      dispatch(getContentViewDetails(cvId));
+      dispatch(getContentViews);
+      setIsOpen(false);
+      setRedirect(true);
+    }
+  }, [JSON.stringify(pollResponse), dispatch, setTaskErrored,
+    setPolling, setIsOpen, setRedirect, pollResponseStatus]);
+
+  if (saving) {
+    return <Loading />;
+  }
+
+  if (polling && pollResponse) {
+    return (
+      <Stack hasGutter>
+        <StackItem span={12} rowSpan={3}>
+          <Bullseye>
+            <div style={{ height: '350px', width: '350px' }}>
+              <ChartDonut
```

Comment:
I feel like this is missing a visual indicator that something is happening.  It's too static.  I wonder if Patternfly has anything that can help..  how would it look if this were superimposed on a [medium Circle skeleton](https://patternfly-react.surge.sh/components/skeleton)?

Also it's too bad that there's no animation transition when the percentages update. But that's probably not worth spending time on.

Reply:
animate={{ duration: 500 }} This adds a bit of animation.